R/utils.R

```diff
@@ -98,22 +98,40 @@ as_binary <- function(x) {
   as.integer(x)
 }
 
-random_string <- function(size = NULL) {
-  # creates a random string using tempfile() which does better to not affect the
-  # random seed
+#' random string function that does not alter the seed.
+#' 
+#' simple wrapper around `stringi::stri_rand_strings()``
+#' 
+#' @param n n
+#' @param length length
+#' @param pattern pattern
+#' @keywords internal
+#' @noRd
+random_string <- function(n = 1, length, pattern = "[A-Za-z0-9]") {
```

Comment:
Add param `keep_seed = FALSE` to skip over all the seed checking/changing?

Reply:
Hm, yes, can do that

Reply:
This is a bit over engineering. Currently it is only used in the style helpers and in the sparkline PR. We could avoid tinkering with saving our seed in the `create_sparkline()` function if we agree that calling that function is not a harmful interaction with the seed. The purpose of this whole PR was to have a function that keeps the seed intact, but still generates various random strings if called multiple times by us, but still keeps the seed intact. Something I don't even think is necessary in the first place 🙈.